internal/venice-common/src/main/java/com/linkedin/venice/helix/ZkRoutersClusterManager.java

```diff
@@ -95,6 +95,11 @@ public void clear() {
     zkClient.unsubscribeStateChanges(zkStateListener);
   }
 
+  public List<String> getLiveRouterInstances() {
+    List<String> list = zkClient.getChildren(getRouterRootPath());
```

Comment:
In theory, the live Router instances can be cached since this class implements `IZkChildListener`, so that we don't need to talk to ZK directly for this.

Reply:
added a cache for router instances.